common/flatpak-transaction.c

```diff
@@ -508,24 +590,67 @@ flatpak_transaction_operation_get_bundle_path (FlatpakTransactionOperation *self
   return self->bundle;
 }
 
+/**
+ * flatpak_transaction_operation_get_commit:
+ * @self: a #FlatpakTransactionOperation
+ *
+ * Gets the commit ID for the operation.
+ *
+ * This information is available when the transaction is resolved,
+ * i.e. when #FlatpakTransaction::ready is emitted.
+ *
+ * Returns: (transfer none): the commit ID
+ */
 const char *
 flatpak_transaction_operation_get_commit (FlatpakTransactionOperation *self)
 {
   return self->resolved_commit;
 }
 
+/**
+ * flatpak_transaction_operation_get_metadata:
+ * @self: a #FlatpakTransactionOperation
+ *
+ * Gets the metadata that will be applicable when the
+ * operation is done.
+ *
+ * This information is available when the transaction is resolved,
+ * i.e. when #FlatpakTransaction::ready is emitted.
+ *
+ * Returns: (transfer none): the metadata #GKeyFile
+ */
 GKeyFile *
 flatpak_transaction_operation_get_metadata (FlatpakTransactionOperation *self)
 {
   return self->resolved_metakey;
 }
 
+/**
+ * flatpak_transaction_operation_get_old_metadata:
+ * @self: a #FlatpakTransactionOperation
+ *
+ * Gets the metadata that will is applicable before the
```

Comment:
This is not right.
old_metadata is the metadata of the current installation, which will be replaced by the one in get_metadata() after the update. You can diff the two to see if we got any new permissions.

Reply:
I was making an educated guess...